src/automation/yaml_generation.py

```diff
@@ -0,0 +1,95 @@
+'''
```

Comment:
might be helpful to add some in line comments throughout this file

Reply:
added